src/inference/gibbs.jl

```diff
@@ -2,7 +2,7 @@
 ### Gibbs samplers / compositional samplers.
 ###
 
-const GibbsComponent = Union{Hamiltonian,MH,ESS,PG}
+const GibbsComponent = Union{Hamiltonian,MH,ESS,PG,GibbsConditional}
```

Comment:
I know it's not part of this PR but IMO we should definitely use traits here.

Reply:
True.  Are traits used anywhere else already?  If not, would make an issue or something to discuss the style or package we should use in the future?

Reply:
Hmm no I don't think so, there were some discussion about refactoring and improving the design of the Gibbs sampler but it might have been part of an older PR. Actually I think for now it could be as simple as defining a `isgibbscomponent(component)` function (or maybe `isgibbsconditional` or some other self-explaining name?) that returns `false` as default and is then checked in the constructor of `Gibbs` since I don't think we dispatch on `GibbsComponent` anywhere in the code. Then every (potentially user-defined) sampler that could be used as part of the Gibbs sampler could implement this function (and potentially in the future a full Gibbs interface).

Reply:
I have made a branch for this solution: https://github.com/phipsgabler/Turing.jl/compare/phg/gibbs_conditionals...phipsgabler:phg/gibbs_component_trait.  But it doesn't feel better.  `GibbsComponent` isn't used anywhere except in the `Gibbs` constructor, but there it adds type safety that is lost with a trait. 

We could add a second `Gibbs` constructor dispatching on the trait, but that would only lead to more confusing type errors IMHO...

Reply:
I still think it makes sense if we check `isgibbscomponent` in an inner constructor of `Gibbs`, even if that's the only place where we (currently) would use it. The main advantage would be that users can allow their custom samplers to be used with the Gibbs algorithm - that can't be done right now since it is not possible to alter or redefine the type union `GibbsComponent` whereas after the change everyone could just implement `isgibbscomponent`.

Reply:
I agree but I think this could be done in an extra PR.

Reply:
Sure, that was my plan, hence the extra branch.

Reply:
Sounds good!